### PR TITLE
Update types for swig-email-templates@7.0.0

### PR DESCRIPTION
**Pull request by himynameisdave**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://github.com/andrewrk/swig-email-templates/commit/ec72a8f95819523021be1485fb76dc7bd246ecae)
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/swig-email-templates/tsconfig.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
         "noImplicitThis": true,
         "strictNullChecks": true,
         "strictFunctionTypes": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/swig-email-templates/tsconfig.json:12]: why this change?
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/swig-email-templates/swig-email-templates-tests.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import EmailTemplates = require('swig-email-templates');
+import EmailTemplates from 'swig-email-templates';
```
Comment (peterblazejewicz) [inline types/swig-email-templates/swig-email-templates-tests.ts:1]: this won't work, once `esModuleInterop` is removed
```suggestion
import EmailTemplates = require('swig-email-templates');
```

File: types/swig-email-templates/index.d.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ declare class EmailTemplates {
     generateSubject(templatePath: string, context: any, cb: (error: any, text: string | null) => void): void;
     rewriteUrls($: JQueryStatic, rewrite: (href: string) => void): void;
     render(templatePath: string, context: any, cb: (error: any, inlinedHTML?: string, text?: string, subject?: string) => void): void;
+    render(templatePath: string, context: any): Promise<{ html?: string; text?: string; subject?: string; }>;
```
[the remark this answers is not in the document]
Comment (himynameisdave) [reply types/swig-email-templates/index.d.ts:28]: Nice, wasn't sure what the correct format here was.

File: types/swig-email-templates/tsconfig.json (hunk 1 above)
Comment (himynameisdave) [reply types/swig-email-templates/tsconfig.json:12]: Had a failing test because I made [this change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61361/files#r926015943). Will revert both 👍

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by himynameisdave** [comment]
Ready to merge
